src/config/utils.ts

```diff
@@ -133,25 +133,57 @@ export function getBoundaryLayers(): BoundaryLayerProps[] {
   const boundaryLayers = Object.values(LayerDefinitions).filter(
     (layer): layer is BoundaryLayerProps => layer.type === 'boundary',
   );
-  if (boundaryLayers.length === 0) {
-    throw new Error(
-      'No boundary layer found. There should be at least one boundary layer defined in layers.json',
-    );
-  }
+
   return boundaryLayers;
 }
 
-export function getBoundaryLayerSingleton(): BoundaryLayerProps {
+export function getDisplayBoundaryLayers(): BoundaryLayerProps[] {
   const boundaryLayers = getBoundaryLayers();
-  const boundaryLayer = boundaryLayers.find(l => l.id === 'admin_boundaries');
+  const boundariesCount = boundaryLayers.length;
 
-  if (!boundaryLayer) {
+  if (boundariesCount === 0) {
     throw new Error(
-      'No admin_boundaries Layer found! There should be exactly one boundary layer defined in layers.json with id admin_boundaries.',
+      'No boundary layer found. There should be at least one boundary layer defined in layers.json',
     );
   }
 
-  return boundaryLayer;
+  if (boundariesCount > 1) {
+    const adminOverrideNames: LayerKey[] = get(
+      appConfig,
+      'adminBoundaryDisplayOverrides',
+      [],
+    );
+
+    const invalidOverrides = adminOverrideNames.filter(
+      id => !boundaryLayers.map(l => l.id).includes(id),
+    );
+
+    if (invalidOverrides.length > 0) {
+      throw new Error(
+        'Some of `adminBoundaryDisplayOverrides` layer Ids are not valid. You must provide valid ids from `layers.json`',
+      );
+    }
+
+    // transoformed overrides to mantain the order of boundaries
+    // since first layer in the list is defined as default
+    const adminOverrideLayers = adminOverrideNames.map(
+      id => boundaryLayers.filter(l => l.id === id)[0],
+    );
+
+    if (adminOverrideLayers.length === 0) {
+      throw new Error(
+        'Multiple boundary layers found. You must provide `adminBoundaryDisplayOverrides` in prism.json',
+      );
+    }
+
+    return adminOverrideLayers;
+  }
+
+  return boundaryLayers;
+}
+
+export function getBoundaryLayerSingleton(): BoundaryLayerProps {
+  return getDisplayBoundaryLayers()[0];
```

Comment:
I don't think that's actually what we want though, no? Don't we want the default admin boundary here? not the for one of the displayed boundaries?

Reply:
I added some comments within the function to clarify a little bit.

Reply:
See my comment above, you actually want to implement a different function, which takes the default admin boundary. Or what you called `primary` above.

@JorgeMartinezG what do you think? See https://github.com/WFP-VAM/prism-frontend/pull/338/files#r782654969 as well for context